### PR TITLE
fix the keytransform datastore's query implementation

### PR DESCRIPTION
**Pull request by Stebalien**
The namespace Datastore shouldn't have to touch the mess with the query. It was also only partially implementing the query logic, passing through filters, etc. without applying the key transform.

Also adds some tests.

### CODE REVIEW

File: keytransform/keytransform.go [hunk 1]
```diff
@@ -78,9 +80,101 @@ func (d *Datastore) Query(q dsq.Query) (dsq.Results, error) {
...
+	for i, o := range child.Orders {
+		switch o.(type) {
+		case dsq.OrderByValue, *dsq.OrderByValue,
+			dsq.OrderByValueDescending, *dsq.OrderByValueDescending:
+			// Key doesn't matter.
+			continue
+		case dsq.OrderByKey, *dsq.OrderByKey,
+			dsq.OrderByKeyDescending, *dsq.OrderByKeyDescending:
+			if orderPreserving {
+				child.Orders = child.Orders[:i+1]
```
Comment (magik6k) [inline keytransform/keytransform.go:126]: I feel like I'm missing something here. Why drop other orders here?

**Review by magik6k — COMMENTED** [review]
Other than that 1 question LGTM

File: keytransform/keytransform.go (hunk 1 above)
Comment (Stebalien) [reply keytransform/keytransform.go:126]: We sort hierarchically. That is, when comparing two entries, we compare with the first "order", then the second order if equal, then the third order if still equal, etc. Given that keys are unique, "order by key" will never return "equal" so there's no point in applying further orders.

Comment (Stebalien) [reply keytransform/keytransform.go:126]: I've added a comment and improved the test coverage more to get codecov to stop complaining.

**Review by magik6k — APPROVED** [review]

**Pull request merged**